2 table.sql

```diff
@@ -0,0 +1,106 @@
+	multi- column constraints
+		name, dob, address and flight # must be unique
+		time arriving must be greater than time departing
+		passport info must be all entered or all blank
+)
+
+*/
+
+use airlineDB
+go 
+drop table if exists dbo.Passenger
+go 
+create table dbo.Passenger(
+	PassengerID int not null identity primary key,
+	FlightNumber  char(6) not null 
+		constraint ck_Passenger_FlightNumber_must_be_FLY_followed_by_3_numbers check (FlightNumber like 'FLY[0-9][0-9][1-9]'),
+	DepartureAirport char(3) not null 
+		constraint ck_Passenger_DepartureAirport_must_be_3_letters check (DepartureAirport like '[a-z][a-z][a-z]'),
+	DepartureCountry varchar(30) not null 
+		constraint ck_Passenger_DepartureCountry_cannot_be_blank check (DepartureCountry > ''),
+	ArrivalAirport char(3) not null
+		constraint ck_Passenger_ArrivalAirport_must_be_3_letters check (ArrivalAirport like '[a-z][a-z][a-z]'),
+	ArrivalCountry varchar(30) not null
+		constraint ck_Passenger_ArrivalCountry_cannot_be_blank check (ArrivalCountry > ''),
+	TimeDeparting datetime not null
+		constraint ck_Passenger_TimeDeparting_cannot_be_more_than_one_year_after_current_date check (TimeDeparting <= dateadd(year, 1, getdate())),
+		--I would add that it cannot be a past date, but the sample data is old
+	TimeArriving datetime not null,
+	PassengerFirstName varchar(30) not null
+		constraint ck_Passenger_PassengerFirstName_cannot_be_blank check (PassengerFirstName > ''),
+	PassengerLastName varchar(30) not null 
+		constraint ck_Passenger_PassengerLastName_cannot_be_blank check (PassengerLastName > ''),
+	PassengerDateOfBirth date not null, 
+	PassengerAddress varchar(150) not null 
+		constraint ck_Passenger_PassengerAddress_cannot_be_blank check (PassengerAddress > ''),
+	PassportNumber char(9) null
+		constraint ck_PassportNumber_must_be_9_numbers_or_null check (PassportNumber like '[1-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]'),
+	PassportIssueDate date null,
+	PassportExpiryDate date null, 
+	Nationality varchar(30) null,
+	--Is there a point in preventing blanks if I allow nulls?
+	constraint u_Passenger_PassengerFirstName_PassengerLastName_PassengerDateOfBirth_PassengerAdress_and_FlightNumber_must_be_unique 
+		unique (PassengerFirstName, PassengerLastName, PassengerDateOfBirth, PassengerAddress, FlightNumber),
+	constraint ck_Passenger_TimeArriving_must_be_greater_than_TimeDeparting 
+		check (TimeArriving > TimeDeparting),
+	constraint ck_Passport_Passport_Information_must_be_all_null_or_all_entered 
+		check ((PassportNumber is null and PassportIssueDate is null and PassportExpiryDate is null and Nationality is null) or (PassportNumber is not null and PassportIssueDate is not null and PassportExpiryDate is not null and Nationality is not null)),
+		--if I allow blanks for passport info, do I have to add a constraint so that everything is null or blank or everything is entered?
+		--or is it enough to just have a constraint that allows all null or all not null?
```

Comment:
This is essentially repeating the previous question in different words. The answer is that null is not blank and vice versa. Allowing or blocking nulls wouldn't have a effect on blanks, and the same the other way around.

Reply:
Great!